jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/EvictingCachingContentFactory.java

```diff
@@ -0,0 +1,165 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.eclipse.jetty.util.NanoTime;
+import org.eclipse.jetty.util.annotation.Name;
+import org.eclipse.jetty.util.thread.Scheduler;
+
+/**
+ * <p>
+ * {@link HttpContent.Factory} implementation of {@link CachingHttpContentFactory} which evicts invalid entries from the cache.
+ * Uses a validationTime parameter to check files when they are accessed and/or sweepDelay parameter for a periodic sweep
+ * for invalid entries in the cache.
+ * </p>
+ * <p>
+ * {@link HttpContent} validation checks are configured through the {@code validationTime} parameter in the constructor.
+ * If an {@link HttpContent} is found to be invalid it will be removed from the cache.
+ * A value of -1 means that the cached {@link HttpContent} will never be checked if it is still valid,
+ * a value of 0 means it is checked on every request,
+ * a positive value indicates the number of milliseconds of the minimum time between validation checks.
+ * </p>
+ * <p>
+ * This also remember a missed entry for the time set by {@code validationTime}ms. After this has
+ * elapsed the entry will be invalid and will be evicted from the cache at the next access.
+ * </p>
+ */
+public class EvictingCachingContentFactory extends CachingHttpContentFactory implements Runnable
```

Comment:
The naming is not exactly correct because it is not this class that does the evicting. That is done by `CachingHttpContentFactory` if the content is not valid.    Perhaps `ValidatingCachingContentFactory`?

Reply:
Done.